nchiki/assignment-2/Ancestors.java

```
@@ -0,0 +1,59 @@
+import java.util.LinkedList;
+import java.util.List;
+
+public class Ancestors {
+
+
+    static Node findANode(Node node, int key) {
+        if(notExists(node)) {
+            return null;
+        }
+
+        else if (node.getValue() == key) {
+            return node;
+        }
+        Node right = findANode(node.getRight(), key);
+        Node left = findANode(node.getLeft(), key);
+
+        if(!notExists(right)) {
+            return right;
+        } else return left;
+    }
+
+    static List<Integer> findAncestors(int key, BinaryTree tree) {
+        Node foundNode = findANode(tree.getRoot(), key);
+
+        if(notExists(foundNode)) {
+            return null;
+        }
+
+        Node parent = foundNode.getParent();
+        List<Integer> values = new LinkedList<>();
+        while (parent != null) {
+            values.add(parent.getValue());
+            parent = parent.getParent();
+        }
+        return values;
+    }
+
+    static int findCommonAncestors(int fstKey, int sndKey, BinaryTree tree) {
+        List<Integer> fstValues = findAncestors(fstKey, tree);
+        List<Integer> sndValues = findAncestors(sndKey, tree);
+
+        if (notExists(fstValues) || notExists(sndValues)) {
+            return Integer.MIN_VALUE;
```

This is very unexpected behavior for someone who would be using this method. Why not to throw an exception instead?